### PR TITLE
update toolchain golang version to 1.18.4

### PR DESCRIPTION
**Pull request by lambotte-lan**
Signed-off-by: Lan Yun <yun.lan@intel.com>

### CODE REVIEW

**Comment by guzongmin** [comment]
The patch looks good to me. Thanks for your contribution.
Two comments:
1. the commit message could be "Update the Golang version to 1.18.4 and use it as the default one."
2. If we could use both 1.16 and 1.18 in the CI test, it could be better. Please update the "https://github.com/occlum/occlum/blob/master/.github/workflows/demo_test.yml" and run one of the Golang tests with both 1.16 and 1.18?

File: tools/toolchains/golang/build.sh [hunk 1]
```diff
@@ -2,6 +2,19 @@
...
+
+if [[ $# -ge 1 ]];then
+case $1 in
+          --help|help)
+               echo  "Usage:$0 [go1.16.3|go1.18.4]"
+               exit
+               ;;
+          "go1.16.3"|"1.16.3"|"v1.16.3")
+               GO_VERSION="1.16.3"
+               ;;
```
Comment (guzongmin) [inline tools/toolchains/golang/build.sh:15]: I think we should add the check about 1.18.4.
"go1.18.4"|"1.18.4"|"v1.18.4")
               GO_VERSION="1.18.4"
               ;;

Comment (lambotte-lan) [reply tools/toolchains/golang/build.sh:15]: ok, I'll add it

File: .github/workflows/demo_test.yml [hunk 2]
```diff
@@ -56,8 +56,11 @@ jobs:
       run: docker exec ${{ github.job }} bash -c "cd /root/occlum/demos/embedded_mode && SGX_MODE=SIM make;
             SGX_MODE=SIM make test"
 
-    - name: Run Golang sqlite test
-      run: docker exec ${{ github.job }} bash -c "cd /root/occlum/demos/golang/go_sqlite/ && SGX_MODE=SIM ./run_go_sqlite_demo.sh"
+    - name: Run Golang v1.16.3 sqlite test
+      run: docker exec ${{ github.job }} bash -c "cd /root/occlum/tools/toolchains/golang && ./build.sh go1.16.3 && cd /root/occlum/demos/golang/go_sqlite/ && SGX_MODE=SIM ./run_go_sqlite_demo.sh"
+
+    - name: Run Golang v1.18.4 sqlite test
+      run: docker exec ${{ github.job }} bash -c "cd /root/occlum/tools/toolchains/golang && ./build.sh && cd /root/occlum/demos/golang/go_sqlite/ && SGX_MODE=SIM ./run_go_sqlite_demo.sh"
```
Comment (guzongmin) [inline .github/workflows/demo_test.yml:63]: I suggest using "./build.sh go1.18.4" to show test motivation clearly.

File: tools/toolchains/golang/build.sh [hunk 3]
```diff
@@ -17,9 +33,9 @@ cd ${BUILD_DIR}
 # Download Golang
 git clone https://github.com/golang/go .
 # Swtich to Golang 1.13.7
```
Comment (guzongmin) [inline tools/toolchains/golang/build.sh:35]: Please update the comments to align with the implementation.

**Review by guzongmin — APPROVED** [review]
LGTM
Thanks for your contribution.

**Pull request merged**